packages/transforms/rename/test/wrapRename.spec.ts

```diff
@@ -481,4 +481,41 @@ describe('rename', () => {
     expect(queryFieldMap.bar).toBeDefined();
     expect(printSchema(newSchema)).toMatchSnapshot();
   });
+
+  it('should track the original name of a renamed field', () => {
```

Comment:
Why do we track renamed fields? In context SDK is passing requests to the transformed schema already.

Reply:
In order to get back to the original field name, which must be used in order to generate valid type declarations: https://github.com/Urigo/graphql-mesh/pull/3414/files#diff-8d4bc1ae7b41e598b72810975c4e0722c93268daa0b53000d30ebf3d5c4d4886R51

Reply:
If we generate types from transformedSchema, we shouldn't need to track these. Maybe I'm missing something here.

Reply:
Oh you're trying to implement another SDK. But can we please seperate it into two different PRs?
- Generate individual types for Incontext SDK with namespaces for each transformed source
- Have another Incontext SDK for the original schema.

Reply:
No I'm not implementing another SDK. This is the same SDK, but in the presence of rename transforms it fails to generate the typescript declarations because the field names do not match. This PR emits source schema type declarations and tracks the renames so that the generate declarations refer to the correct source schema types.

Reply:
To clarify (as original reported in issue #3367 ): if you rename fields in the source schema, and then filter those fields out in the root filter, the `.mesh/index.d.ts` type declarations are not emitted. The sandbox that demonstrates it is here: https://codesandbox.io/s/condescending-swanson-9n0ux?file=/.mesh/schema.graphql

Just run `yarn build` and observe that the declarations are not emitted. If you remove the root filter, they are. If you modify the artifact generation so it writes the "intermediate" TS to disk, when you look inside you'll see the field name mismatch.